modules/virt-querying-metrics.adoc

```diff
@@ -0,0 +1,98 @@
+// Module included in the following assemblies:
+//
+// * virt/logging_events_monitoring/virt-prometheus-queries.adoc
+
+[id="virt-querying-metrics_{context}"]
+= Virtualization metrics
+
+The following metric descriptions include example Prometheus Query Language (PromQL) queries.
+
+[NOTE]
+====
+These metrics are not an API and might change between versions.
+====
+
+
+[id="virt-promql-vcpu-metrics_{context}"]
+== vCPU metrics
+
+`kubevirt_vmi_vcpu_wait_seconds`::
+Returns the wait time (in seconds) for a virtual machine's vCPU.
+
+A value above '0' means that the vCPU wants to run, but the host scheduler can not run it yet. This can indicate that there is an issue with Input/Output.
+
+.Example query
+[source,promql]
+----
+topk(3, sum by (name, namespace) (round(irate(kubevirt_vmi_vcpu_wait_seconds[6m]), 0.1))) > 0
+----
+The above query returns the top 3 VMs waiting for I/O at every given moment in time over the time period.
+
+[id="virt-promql-network-metrics_{context}"]
+== Network metrics
+
+`kubevirt_vmi_network_receive_bytes_total`::
+Returns the total amount of traffic received (in bytes) on the virtual machine's network.
+
+`kubevirt_vmi_network_transmit_bytes_total`::
+Returns the total amount of traffic transmitted (in bytes) on the virtual machine's network.
+
+These queries can be used to identify virtual machines that are saturating the network.
+
+.Example query
+[source,promql]
+----
+topk(3, sum by (name, namespace) (round(irate(kubevirt_vmi_network_receive_bytes_total[6m]), 0.1)) + sum by (name, namespace) (round(irate(kubevirt_vmi_network_transmit_bytes_total[6m]) , 0.1))) > 0
+----
+The above query returns the top 3 VMs transmitting the most network traffic at every given moment in time over a six-minute time period.
+
+[id="virt-promql-storage-metrics_{context}"]
+== Storage metrics
+
+`kubevirt_vmi_storage_read_traffic_bytes_total`::
+Returns the total amount (in bytes) of the virtual machine's storage-related traffic. 
+
+`kubevirt_vmi_storage_write_traffic_bytes_total`::
+Returns the total amount of storage writes (in bytes) of the virtual machine's storage-related traffic.
+
+These queries can be used to identify virtual machines that are writing large amounts of data.
+
+.Example query
+[source,promql]
+----
+topk(3, sum by (name, namespace) (round(irate(kubevirt_vmi_storage_read_traffic_bytes_total[6m]), 0.1)) + sum by (name, namespace) (round(irate(kubevirt_vmi_storage_write_traffic_bytes_total[6m]), 0.1))) > 0
```

Comment:
Would it make sense to break down this long query into multiple lines? Same applies to the other two below.

Reply:
Great idea - and yes. Turns out the UI is happy with linebreaks as well so multiple lines copy+paste just fine.